### PR TITLE
♻️ Refactor Functions

### PR DESCRIPTION
**Pull request by athul**
Make use of `ES6+` constructs
- Arrow functions
- Template literals
- Object destructuring (concise code)

### CODE REVIEW

File: assets/js/curlparser.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import * as querystring from "querystring";
  * output this: 'msg1=value1&msg2=value2'
  * @param dataArguments
  */
-function joinDataArguments(dataArguments) {
+ let joinDataArguments=(dataArguments)=> {
```
Comment (NBTX) [inline assets/js/curlparser.js]: Can you include a space both either side of the function name and equals sign, and the closing parameter bracket and opening function brace:

**Before:**
```javascript
let joinDataArguments=(dataArguments)=> {
```

**After:**
```javascript
let joinDataArguments = (dataArguments) => {
```

**Comment by athul** [comment]
@NBTX Done ✨ ✨

File: assets/js/curlparser.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import * as querystring from "querystring";
  * output this: 'msg1=value1&msg2=value2'
  * @param dataArguments
  */
-function joinDataArguments(dataArguments) {
+let joinDataArguments = (dataArguments) => {
```
Comment (jamesgeorge007) [inline assets/js/curlparser.js]:
```suggestion
const joinDataArguments = (dataArguments) => {
```

File: components/collections/editRequest.vue [hunk 3]
```diff
@@ -93,7 +93,7 @@ export default {
     };
   },
   watch: {
-    "requestUpdateData.collectionIndex": function resetFolderIndex() {
+    "requestUpdateData.collectionIndex": ()=> { //resetFolderIndex Function
```
Comment (jamesgeorge007) [inline components/collections/editRequest.vue]: An arrow function wouldn't bind this to the Vue instance. Please revert this change.

File: components/collections/saveRequestAs.vue [hunk 4]
```diff
@@ -119,7 +119,7 @@ export default {
       this.$data.requestData.folderIndex = undefined;
       this.$data.requestData.requestIndex = undefined;
     },
-    "requestData.folderIndex": function resetRequestIndex() {
+    "requestData.folderIndex": ()=> {
```
Comment (jamesgeorge007) [inline components/collections/saveRequestAs.vue]: Same as above

File: components/collections/editRequest.vue (hunk 3 above)
Comment (athul) [reply components/collections/editRequest.vue]: Woah @jamesgeorge007 You're awesome mate

File: pages/realtime.vue [hunk 5]
```diff
@@ -432,8 +432,8 @@ export default {
       this.sse.close();
     }
   },
-  updated: function() {
-    this.$nextTick(function() {
+  updated: ()=> {
```
Comment (jamesgeorge007) [inline pages/realtime.vue]: Don't use arrow functions on an options property

Comment (athul) [reply pages/realtime.vue]: Okay, I will revert this back . I'm really sorry for the Trouble Caused

**Comment by athul** [comment]
@jamesgeorge007 Done I guess 😂

**Comment by athul** [comment]
I guess that most of the Functions refactored on .vue files and I guess thats why all of em are failing(tests). I think this should be closed as of now and I'll create a new PR with just the .js files refactored.

How does that sound?

**Comment by athul** [comment]
What do you say @liyasthomas @NBTX @jamesgeorge007 
Close it, if you feel like this should be closed

**Comment by liyasthomas** [comment]
@athul no need. I'm quite busy with office stuffs rn, will be able to make this workable with some tweaks.

File: components/collections/editRequest.vue [hunk 6]
```diff
@@ -93,7 +93,7 @@ export default {
     };
   },
   watch: {
-    "requestUpdateData.collectionIndex": function resetFolderIndex() {
+    "requestUpdateData.collectionIndex":function resetFolderIndex() { //resetFolderIndex Function
```
Comment (jamesgeorge007) [inline components/collections/editRequest.vue]:
```suggestion
    "requestUpdateData.collectionIndex": function resetFolderIndex() {
```

Comment (jamesgeorge007) [reply components/collections/editRequest.vue]: FYI this is a commit suggestion, just click on `commit suggestion` and you can commit the suggested change with ease.

File: pages/index.vue [hunk 7]
```diff
@@ -1989,7 +1989,7 @@ export default {
   },
   mounted() {
     this.observeRequestButton();
-    this._keyListener = function(e) {
+    this._keyListener = (e)=> {
```
Comment (athul) [inline pages/index.vue]: @jamesgeorge007 What abt these arrows? Will they work correctly?

Comment (jamesgeorge007) [reply pages/index.vue]: I guess they're gonna introduce scope resolution ambiguities

Comment (athul) [reply pages/index.vue]: English please 😅

Comment (liyasthomas) [reply pages/index.vue]: We can't use arrow operator on any functions involving `this` since it's scope is targeted to that function, not global vars.

Comment (athul) [reply pages/index.vue]: 😿

**Comment by liyasthomas** [comment]
@jamesgeorge007 please take a look at this PR. Contribute to this pull if there's any refactoring left.

File: assets/js/curlparser.js [hunk 8]
```diff
@@ -103,9 +103,7 @@ function parseCurlCommand(curlCommand) {
   }
   if (cookieString) {
     let cookieParseOptions = {
```
Comment (jamesgeorge007) [inline assets/js/curlparser.js]:
```suggestion
     const cookieParseOptions = {
```

File: assets/js/curlparser.js [hunk 9]
```diff
@@ -175,7 +173,7 @@ function parseCurlCommand(curlCommand) {
 
   urlObject.search = null; // Clean out the search/query portion.
   let request = {
```
Comment (jamesgeorge007) [inline assets/js/curlparser.js]:
```suggestion
   const request = {
```
